contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -803,20 +830,12 @@ contract AerodromeAMOStrategy is InitializableAbstractStrategy {
     }
 
     /**
-     * @notice Withdraw an `amount` of assets from the platform and
-     *         send to the `_recipient`.
-     * @param _recipient  Address to which the asset should be sent
-     * @param _asset      WETH address
-     * @param _amount     Amount of WETH to withdraw
+     * @dev This function removes the appropriate amount of liquidity to assure that the required
+     * amount of WETH is available on the contract
+     *
+     * @param _amount  WETH balance required on the contract
      */
-    function withdraw(
-        address _recipient,
-        address _asset,
-        uint256 _amount
-    ) external override onlyVault nonReentrant {
-        require(_asset == WETH, "Unsupported asset");
-        require(_recipient == vaultAddress, "Only withdraw to vault allowed");
-
+    function _assureWETHBalance(uint256 _amount) internal {
         uint256 _wethBalance = IERC20(WETH).balanceOf(address(this));
         if (_wethBalance < _amount) {
```

Comment:
⚪️ Can flip this, and change this to fast return, now that this if statement is the whole function?

Reply:
nice short circuit spot, changed